OADB/AliPhysicsSelection.cxx

```diff
@@ -253,61 +254,13 @@ UInt_t AliPhysicsSelection::CheckTriggerClass(const AliVEvent* event, const char
 
 //______________________________________________________________________________
 Bool_t AliPhysicsSelection::EvaluateTriggerLogic(const AliVEvent* event, AliTriggerAnalysis* triggerAnalysis, const char* triggerLogic, Bool_t offline){
-  // evaluates trigger logic. If called with no event pointer/triggerAnalysis pointer, it just caches the tokens
-  // Fills the statistics histogram, if booked at row i
-  TString trigger(triggerLogic);
-  
-  // add space after each token (to use ReplaceAll later)
-  fRegexp->Substitute(trigger, "$1 ", "g");
-  
-  while (1) {
-    AliDebug(AliLog::kDebug, trigger.Data());
-    
-    TArrayI pos;
-    Int_t nMatches = fRegexp->Match(trigger, "", 0, 2, &pos);
-    
-    if (nMatches <= 0) break;
-    
-    TString token(trigger(pos[0], pos[1]-pos[0]+1));
-
-    TParameter<Int_t>* param = dynamic_cast<TParameter<Int_t> *>(fCashedTokens->FindObject(token));
-    if (!param) {
-      TInterpreter::EErrorCode error;
-      Int_t bit = gInterpreter->ProcessLine(Form("AliTriggerAnalysis::k%s;", token.Data()), &error);
-      
-      if (error > 0) AliFatal(Form("Trigger token %s unknown", token.Data()));
-      
-      param = new TParameter<Int_t>(token, bit);
-      fCashedTokens->Add(param);
-      AliDebug(AliLog::kDebug, "Added token");
-    }
-    
-    Long64_t bit = param->GetVal();
-    
-    AliDebug(AliLog::kDebug, Form("Tok %d %d %s %lld", pos[0], pos[1], token.Data(), bit));
-    
-    if (offline) 
-      bit |= AliTriggerAnalysis::kOfflineFlag;
-    
-    if(event && triggerAnalysis) {
-      trigger.ReplaceAll(token, Form("%d", triggerAnalysis->EvaluateTrigger(event, (AliTriggerAnalysis::Trigger) bit)));
-    }
+  // Do we already have a function for this trigger logic? Add to hashmap if not
+  if (fTriggerToFunction->count(triggerLogic) == 0) {
```

Comment:
Thats a neat way! I'll fix this if we go this route! I just relied on hashmap look-ups to be dirt cheap anyways.